types/basetypes/float64_type.go

```diff
@@ -9,16 +9,18 @@ import (
 	"math"
 	"math/big"
 
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+
 	"github.com/hashicorp/terraform-plugin-framework/attr"
 	"github.com/hashicorp/terraform-plugin-framework/attr/xattr"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/path"
-	"github.com/hashicorp/terraform-plugin-go/tftypes"
 )
 
 // Float64Typable extends attr.Type for float64 types.
 // Implement this interface to create a custom Float64Type type.
 type Float64Typable interface {
+	//nolint:staticcheck // xattr.TypeWithValidate is deprecated, but we still need to support it.
```

Comment:
I was wondering about this. For float64 and int64 I was thinking that this may not be necessary as the calls to  `<Float64|Int64>Type.ValueFromTerraform` ensure that the `tftypes.Value` supplied can be represented as 64-bit floating point, and integer values, respectively. The constructors on the corresponding value types only accept 64-bit floating point, and integer values too (e.g., `NewFloat64Value`). So it doesn't seem that there is a way to generate a float64 or int64 value type that would have a value that couldn't be represented as a 64-bit floating point, or integer value, respectively.

I'll create a follow-up issue for list, map, set. Let me know if you think float64 and int64 should be included and we can discuss this further.